node_modules/oae-principals/tests/test-email.js

```diff
@@ -0,0 +1,280 @@
+
+        // Drain the email queue
+        EmailTestsUtil.clearEmailCollections(callback);
+    });
+
+    describe('Email tokens', function() {
+
+        describe('Events', function() {
+            /**
+             * Test that verifies that an email token is sent when creating a user account through the REST api
+             */
+            it('verify a verification email is sent when creating a user account through the REST api', function(callback) {
+                // Disable recaptcha so we can create a user
+                ConfigTestUtil.updateConfigAndWait(camAdminRestContext, null, {'oae-principals/recaptcha/enabled': false}, function(err) {
+                    assert.ok(!err);
+
+                    var username = TestsUtil.generateTestUserId();
+                    var email = TestsUtil.generateTestEmailAddress(username);
+                    RestAPI.User.createUser(anonymousRestContext, username, 'password', 'Test User', {'email': email}, function(err) {
+                        assert.ok(!err);
+                    });
+
+                    EmailAPI.once('debugSent', function(message) {
+                        // Assert we've sent an email to the correct user
+                        assert.strictEqual(message.to.length, 1);
+                        assert.strictEqual(message.to[0].address, email);
+
+                        // Assert that the email contains both plain text and html
+                        assert.ok(message.html);
+                        assert.ok(message.text);
+
+                        // Assert a URL is included that can be used to verify the email address
+                        assert.ok(message.html.match(/\?token=[a-zA-Z0-9]{32}/));
+                        assert.ok(message.text.match(/\?token=[a-zA-Z0-9]{32}/));
+
+                        // Assert the email is not verified yet
+                        var restContext = TestsUtil.createTenantRestContext(global.oaeTests.tenants.cam.host, username, 'password');
+                        RestAPI.User.getMe(restContext, function(err, me) {
+                            assert.ok(!err);
+                            assert.strictEqual(me.email, email);
+                            assert.strictEqual(me.emailVerified, false);
+
+                            // Enable re-captcha again
+                            ConfigTestUtil.updateConfigAndWait(camAdminRestContext, null, {'oae-principals/recaptcha/enabled': true}, function(err) {
+                                assert.ok(!err);
+                                return callback();
+                            });
+                        });
+                    });
+                });
+            });
+
+            /**
+             * Test that verifies that an email token is sent when a user updates his email address
+             */
+            it('verify a verification email is sent when a user updates his email address', function(callback) {
```

Comment:
Also, what about the scenario where a user in the Cambridge tenant updates his email address to `*@gatech.edu`. Is that something we want to allow for? I guess it would be okay as they'd need to validate the address.